backend/app/api/endpoints/groups.py

```
@@ -283,6 +286,37 @@ def update_member_role_endpoint(
 
 
+@router.put(
+    "/{group_name:path}/members/batch/roles",
+    response_model=GroupMemberBatchUpdateResponse,
+)
+def update_member_roles_batch_endpoint(
+    group_name: str = Path(
+        ..., description="Group name (may contain slashes for subgroups)"
+    ),
+    batch_update: GroupMemberBatchUpdateRequest = None,
+    current_user: User = Depends(get_current_user),
```

Ruff (0.15.6)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)